demo/app/ios/Runner/OpenID4CI.swift

```diff
@@ -31,14 +31,53 @@ public class OpenID4CI {
         self.initiatedInteraction = Openid4ciNewInteraction(args, opts, nil)!
     }
     
+    func checkFlow() throws -> String {
+        let issuerCapabilities = initiatedInteraction.issuerCapabilities()
+        if ((issuerCapabilities!.authorizationCodeGrantTypeSupported())){
+            return "auth-code-flow"
+        }
+        if ((issuerCapabilities!.preAuthorizedCodeGrantTypeSupported())){
+           return "preauth-code-flow"
+        }
+        return ""
+    }
+    
+    func getAuthorizationLink() throws -> String {
+        let issuerCapabilities = initiatedInteraction.issuerCapabilities()
+        if !(issuerCapabilities?.authorizationCodeGrantTypeSupported())! {
+            // This code example isn't applicable. See the authorization code flow example instead.
+            return "Not implemented"
+        }
+        
+        let scopes = ApiStringArray()
+        scopes!.append("openid")!.append("profile")
+        // TODO #423 Read withScopes and redirect uri from flutter enviornment. Replace these with approriate values as of now.
+        let authorizationLink = initiatedInteraction.createAuthorizationURL(withScopes: "clientID", redirectURI: "redirect URI", scopes: scopes, error: nil)
+        
+        return authorizationLink
+    }
+
+    
+    
     func pinRequired() throws -> Bool {
+        let issuerCapabilities = initiatedInteraction.issuerCapabilities()
+        if  !issuerCapabilities!.preAuthorizedCodeGrantTypeSupported() {
+            // This code example isn't applicable. See the authorization code flow example instead.
+            
+            return false
```

Comment:
Need to throw an error here

Reply:
(if pre-auth code flow isn't support, then `pinRequired` doesn't make sense)

Reply:
error handling I am doing separately in this pr https://github.com/trustbloc/wallet-sdk/issues/426 as I have to catch properly in the flutter layer and show error too